api/v1alpha1/clusterextension_types.go

```diff
@@ -46,6 +46,104 @@ const (
 
 // ClusterExtensionSpec defines the desired state of ClusterExtension
 type ClusterExtensionSpec struct {
+	// source is a required field which selects the install source, such as a
+	// Catalog or another source such as a Bundle (support TBA).
+	//
+	// source must contain the following two fields:
+	//   - sourceType: must be a supported string value; "Catalog" is currently the only accepted value
+	//   - <source>:  object with a key name matching the chosen sourceType; only one such field may be defined
+	//
+	// For example, sourceType: "Catalog" requires the catalog field to
+	// be defined, whereas sourceType: "Bundle" (not implemented) would require
+	// the bundle field to be defined instead.
+	//
+	// The catalog field contains references specific to catalogs, such as the
+	// packageName (required), version and channel.
```

Comment:
We can probably remove this block of text, I think the descriptions on the individual fields is sufficient

Reply:
removed per request.